docs/source/common/trainer.rst

```diff
@@ -1532,7 +1532,8 @@ How often within one training epoch to check the validation set.
 Can specify as float or int.
 
 - pass a ``float`` in the range [0.0, 1.0] to check after a fraction of the training epoch.
-- pass an ``int`` to check after a fixed number of training batches.
+- pass an ``int`` to check after a fixed number of training batches. An ``int`` value can only be higher than the number of training
+  batches when ``check_val_every_n_epoch=None``, which validates after every ``N`` training batches across complete training.
```

Comment:
Maybe something like this?

```suggestion
  batches when ``check_val_every_n_epoch=None``, which validates every ``N`` training batches across epochs.
```

Reply:
across epochs might be incorrect if we consider max_steps too. Any better recommendation?

Also PR is incomplete right now. The progress bars are broken for some edge cases. Need to update them.

Reply:
Is this still blocked?